101-machine-learning-create/azuredeploy.json

```diff
@@ -2,257 +2,641 @@
-          "apiVersion": "2018-05-01-preview",
-          "name": "[parameters('applicationInsightsName')]",
-          "location": "[if(or(equals(parameters('location'),'eastus2'), equals(parameters('location'),'westcentralus')),'southcentralus',parameters('location'))]",
-          "kind": "web",
-          "properties": {
-              "Application_Type": "web"
-          }
-      },
-      {
-          "type": "Microsoft.MachineLearningServices/workspaces",
-          "apiVersion": "2020-03-01",
-          "name": "[parameters('workspaceName')]",
-          "location": "[parameters('location')]",
-          "dependsOn": [
-              "[variables('storageAccount')]",
-              "[variables('keyVault')]",
-              "[variables('applicationInsights')]"
-          ],
-          "identity": {
-              "type": "systemAssigned"
-          },
-          "sku": {
-              "tier": "[parameters('sku')]",
-              "name": "[parameters('sku')]"
-          },
-          "properties": {
-              "friendlyName": "[parameters('workspaceName')]",
-              "storageAccount": "[variables('storageAccount')]",
-              "keyVault": "[variables('keyVault')]",
-              "applicationInsights": "[variables('applicationInsights')]",
-              "containerRegistry": "[if(empty(parameters('containerRegistry')), json('null'), parameters('containerRegistry'))]",
-              "adbWorkspace": "[if(empty(parameters('adbWorkspace')), json('null'), parameters('adbWorkspace'))]",
-              "encryption": {
-                  "status": "[parameters('encryption_status')]",
-                  "keyVaultProperties": {
-                      "keyVaultArmId": "[parameters('cmk_keyvault')]",
-                      "keyIdentifier": "[parameters('resource_cmk_uri')]"
+                  "privateEndpointNicResourceId": {
+                    "value": "[if(equals(parameters('privateEndpointType'), 'AutoApproval'),
+                                    reference(resourceId('Microsoft.Network/privateEndpoints',concat(parameters('workspaceName'), '-PrivateEndpoint'))).networkInterfaces[0].id,
+                                    resourceId('Microsoft.Network/privateEndpoints',concat(parameters('workspaceName'), '-PrivateEndpoint')).networkInterfaces[0].id)]"
+
+                  },
+                  "nicRecordsTemplateUri": {
+                    "value": "https://network.hosting.portal.azure.net/network/Content/4.13.392.925/DeploymentTemplates/PrivateDnsForPrivateEndpointNic.json"
+                  },
+                  "ipConfigRecordsTemplateUri": {
+                    "value": "https://network.hosting.portal.azure.net/network/Content/4.13.392.925/DeploymentTemplates/PrivateDnsForPrivateEndpointIpConfig.json"
+                  },
+                  "uniqueId": {
+                    "value": "[variables('privateDnsGuid')]"
+                  },
+                  "existingRecords": {
+                    "value": {}
```

Comment:
not sure I follow the intent with these deployments, since this is always empty...

Reply:
Removed